lib/galaxy/files/sources/_pyfilesystem2.py

```diff
@@ -34,7 +34,11 @@ def list(self, path="/", recursive=False, user_context=None):
                     res.extend(map(to_dict, files))
                 return res
             else:
-                res = h.scandir(path, namespaces=['details'])
+                try:
+                    res = h.scandir(path)
+                except Exception:
+                    log.warning("Something in remote files listing went wrong.")
+                    return []
```

Comment:
Thanks so much for the fixes, this stuff clearly requires polish and this PR is a great step in that direction.

For the exception handling though, I think this is probably an inappropriate place to catch and swallow the exception. I think it needs to percolate up the stack and get back to the user somehow. I understand this harder though and requires GUI changes. Can you rebase config changes out of this and we can hack on the error handling in a separate PR?

I've started with https://github.com/galaxyproject/galaxy/pull/11581 - which I think should provide similar feedback in the logs to this and handles exceptions in the above ``h.walk`` path also. Any chance you can try that branch out and provide feedback?

Reply:
Ah yes, your pr looks better, I'll have a look and test it :)
I've just rebased this one